integrations/coinbase-integration/src/main/java/org/dash/wallet/integration/coinbase_integration/ui/CoinbaseBuyDashOrderReviewFragment.kt

```diff
@@ -81,18 +79,31 @@ class CoinbaseBuyDashOrderReviewFragment : Fragment(R.layout.fragment_coinbase_b
         super.onViewCreated(view, savedInstanceState)
 
         requireActivity().onBackPressedDispatcher.addCallback(viewLifecycleOwner) {
-            analyticsService.logEvent(AnalyticsConstants.Coinbase.BOTTOM_BACK_TO_ENTER_AMOUNT, bundleOf())
+            viewModel.logEvent(AnalyticsConstants.Coinbase.BUY_QUOTE_ANDROID_BACK)
             findNavController().popBackStack()
         }
 
         binding.toolbar.setNavigationOnClickListener {
-            analyticsService.logEvent(AnalyticsConstants.Coinbase.TOP_BACK_TO_ENTER_AMOUNT, bundleOf())
+            viewModel.logEvent(AnalyticsConstants.Coinbase.BUY_QUOTE_TOP_BACK)
             findNavController().popBackStack()
         }
 
         binding.cancelBtn.setOnClickListener {
-            analyticsService.logEvent(AnalyticsConstants.Coinbase.CANCEL_DASH_PURCHASE, bundleOf())
-            safeNavigate(CoinbaseBuyDashOrderReviewFragmentDirections.confirmCancelBuyDashTransaction())
+            viewModel.logEvent(AnalyticsConstants.Coinbase.BUY_QUOTE_CANCEL)
+            val dialog = AdaptiveDialog.simple(
+                getString(R.string.cancel_transaction),
+                getString(R.string.no_keep_it),
+                getString(R.string.yes_cancel)
+            )
+            dialog.isCancelable = false
+            dialog.show(requireActivity()) { result ->
+                if (result == true) {
+                    viewModel.logEvent(AnalyticsConstants.Coinbase.BUY_QUOTE_CANCEL_YES)
```

Comment:
Another disadvantage of navigating to simple dialogs - hard to extend the logic of positive/negative action.